components/script/script_thread.rs

```diff
@@ -1218,24 +1256,22 @@ impl ScriptThread {
         let mut dom_tree_size = 0;
         let mut reports = vec![];
 
-        if let Some(root_context) = self.browsing_context.get() {
-            for it_context in root_context.iter() {
-                let current_url = it_context.active_document().url().to_string();
+        for document in self.documents.borrow().iter() {
+            let current_url = document.url().to_string();
```

Comment:
You might be able to get by with `let current_url = document.url().as_str();` here?